src/commands/appSettings/connectionSettings/eventHubs/EventHubsConnectionPromptStep.ts

```diff
@@ -22,65 +22,63 @@ export class EventHubsConnectionPromptStep<T extends IEventHubsConnectionWizardC
     public async prompt(context: T): Promise<void> {
         const connectEventNamespaceButton: MessageItem = { title: localize('connectEventHubsNamespace', 'Connect Event Hub Namespace') };
         const useEmulatorButton: MessageItem = { title: useEmulator };
-        const skipForNowButton: MessageItem = { title: skipForNow };
 
         const message: string = localize('selectEventHubsNamespace', 'In order to proceed, you must connect an event hub namespace for internal use by the Azure Functions runtime.');
 
         const buttons: MessageItem[] = [connectEventNamespaceButton, useEmulatorButton];
 
-        if (!this._options?.suppressSkipForNow) {
-            buttons.push(skipForNowButton);
+        const eventHubConnection: string | undefined = await getLocalSettingsConnectionString(context, ConnectionKey.EventHubs, context.projectPath);
+        if (!!eventHubConnection && !localEventHubsEmulatorConnectionRegExp.test(eventHubConnection)) {
+            return undefined;
         }
 
         const result: MessageItem = await context.ui.showWarningMessage(message, { modal: true }, ...buttons);
         if (result === connectEventNamespaceButton) {
-            context.eventHubConnectionType = ConnectionType.Azure;
-        } else if (result === useEmulatorButton) {
-            // 'NonAzure' will represent 'Emulator' in this flow
-            context.eventHubConnectionType = ConnectionType.NonAzure;
+            context.eventHubsConnectionType = ConnectionType.Azure;
         } else {
-            context.eventHubConnectionType = ConnectionType.None;
+            context.eventHubsConnectionType = ConnectionType.Emulator;
         }
 
-        context.telemetry.properties.eventHubConnectionType = context.eventHubConnectionType;
+        context.telemetry.properties.eventHubConnectionType = context.eventHubsConnectionType;
     }
 
-    public shouldPrompt(context: T): boolean {
-        if (this._options?.preselectedConnectionType) {
-            context.eventHubConnectionType = this._options.preselectedConnectionType;
+    public async configureBeforePrompt(context: T): Promise<void> {
+        if (this._options?.preselectedConnectionType === ConnectionType.Azure || this._options?.preselectedConnectionType === ConnectionType.Emulator) {
```

Comment:
This used to just always set `context.eventHubConnectionType = this._options.preselectedConnectionType`. Did we need to change this just because there is the `ConnectionType.Custom` now?